genai-perf/genai_perf/config/generate/search_parameters.py

```diff
@@ -64,21 +47,20 @@ def __init__(
         is_composing_model: bool = False,
     ):
         self._subcommand = subcommand
-        if subcommand == Subcommand.OPTIMIZE:
-            self._config = config.optimize
-        elif subcommand == Subcommand.ANALYZE:
-            self._config = config.analyze  # type: ignore
 
-        # TODO: OPTIMIZE
-        # self._supports_model_batch_size = model.supports_batching()
+        if subcommand == Subcommand.ANALYZE:
+            self._config = config.analyze
+        else:
+            raise GenAIPerfException(f"Subcommand {subcommand} not supported")
```

Comment:
I am assuming this will never be hit in practice?
This is just here to guard against it and throw a warning for us to debug. Is that correct?

Reply:
Yes, this can never be hit. I suppose this could also be an assert to ensure that the subcommand is analyze. What's your preference?

Reply:
This is fine. They are functionally equal